src/hardware/setup.rs

```diff
@@ -688,17 +688,26 @@ pub fn setup(
 
         unsafe { ethernet::enable_interrupt() };
 
+        // Configure IP address according to DHCP socket availability
+        let ip_addrs: smoltcp::wire::IpAddress = option_env!("STATIC_IP")
+            .unwrap_or("UNSPECIFIED")
+            .parse()
+            .unwrap();
+
         // Note(unwrap): The hardware configuration function is only allowed to be called once.
         // Unwrapping is intended to panic if called again to prevent re-use of global memory.
         let store =
             cortex_m::singleton!(: NetStorage = NetStorage::default()).unwrap();
 
-        store.ip_addrs[0] = smoltcp::wire::IpCidr::new(
-            smoltcp::wire::IpAddress::Ipv4(
-                smoltcp::wire::Ipv4Address::UNSPECIFIED,
+        store.ip_addrs[0] = match ip_addrs.is_unspecified() {
```

Comment:
```suggestion
        store.ip_addrs[0] = smoltcp::wire::IpCidr::new(option_env!("STATIC_IP").unwrap_or("0.0.0.0").parse().unwrap(), 24);
```

Reply:
Didn't follow the suggestion exactly but used 26 bit masks for both specified and unspecified IP cases. Also fixed an `UNSPECIFIED` mistake I made in the previous commit and tested everything on Stabilizer without issues